app/src/main/java/com/epfl/drawyourpath/database/FirebaseDatabaseUtils.kt

```diff
@@ -122,6 +125,63 @@ object FirebaseDatabaseUtils {
         )
     }
 
+    /**
+     * Helper function to obtain the posts of a tournament from the database
+     * @param data the data snapshot containing the posts
+     * @return a list containing the posts of the tournament
+     */
+    fun transformPostList(data: DataSnapshot?): List<TournamentPost> {
+        return data?.children?.mapNotNull {
+            transformPost(it)
+        } ?: emptyList()
+    }
+
+    /**
+     * Helper function to obtain a post from a database snapshot
+     * @param data the data snapshot containing the post
+     * @return the post, or null if an error occurred
+     */
+    fun transformPost(data: DataSnapshot?): TournamentPost? {
+        val userId = data?.child("userId")?.value as String?
+        val run = transformRun(data?.child("run"))
+        val votes = getNumber(data?.child("votes"))?.toInt()
+        val date = transformLocalDateTime(data?.child("date"))
+        // Unchecked cast here but should work without problem
+        val usersVotes =
+            (data?.child("usersVotes")?.value ?: emptyMap<String, Int>()) as Map<String, Int>
+
+        if (userId == null || run == null || votes == null || date == null) {
+            Log.e(this::class.java.name, "TournamentPost had null values")
+            return null
+        }
+
+        return TournamentPost(userId, run, votes, date, usersVotes.toMutableMap())
+    }
+
+    /**
+     * Helper function to obtain an object LocalDateTime from a data snapshot
+     * @param data the data snapshot containing the LocalDateTime
+     * @return the LocalDateTime, or null if an error occurred
```

Comment:
Suggestion: Consider using an exception instead of returning null to indicate errors